project/src/main/java/com/google/sps/store/SearchStore.java

```diff
@@ -59,27 +66,22 @@ private void addKeywordsInTextToIndex(String eventId, String text, float weight)
 
     for (Keyword keyword : keywords) {
       float keywordRank = 0;
-      Optional<EventResult> existingResult =
-          keywordToEventResults.get(keyword.getName()).stream()
-              .filter(result -> result.getEventId().equals(eventId))
-              .findFirst();
-      if (existingResult.isPresent()) {
-        keywordRank += existingResult.get().getRanking();
-        keywordToEventResults.remove(keyword.getName(), existingResult.get());
+      Float existingRanking = keywordToRanking.get(keyword.getName());
+      if (existingRanking != null) {
+        keywordRank += existingRanking;
       }
-      keywordRank += weight * keyword.getRelevance();
-      keywordToEventResults.put(keyword.getName(), new EventResult(eventId, keywordRank));
+      keywordRank = Float.sum(keywordRank, weight * keyword.getRelevance());
+      keywordToRanking.put(keyword.getName(), keywordRank);
     }
   }
 
   /**
-   * Gets search results for the given keyword.
-   * @param keyword keyword in search
-   * @return list of event IDs in decreasing order of ranking as search result
+   * Gets event searchResults for the given keyword search.
+   *
+   * @param keyword keyword to search
+   * @return list of events in descending order of ranking as search result
    */
-  public List<String> getSearchResults(String keyword) {
-    List<EventResult> results = keywordToEventResults.get(keyword);
-    Collections.sort(results, EventResult.ORDER_BY_RANKING_DESC);
-    return results.stream().map(EventResult::getEventId).collect(Collectors.toList());
+  public static List<Event> getSearchResults(String keyword) {
```

Comment:
Looking ahead, you'll probably also want to call part of Stephanie's function to rank the events (so you would return all of the correct events, but they're also in a ranked order)

Reply:
I see the thinking but for user manual search, there is a different ranking algorithm.